drivers/hwinfo/Kconfig.ambiq

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2024 Ambiq
```

Comment:
as these files are new, wouldn't it be better to use the recommended Copyright notice:
https://docs.zephyrproject.org/latest/contribute/guidelines.html#copyrights-notices

Reply:
I thought it would make sense to preserve the copyright from the original change that added support.

@alexander-waechter - do you have any preference?

Reply:
Not really. I don't mind if I'm in or not 😁

Reply:
Fixed!